src/shuffle_docids.cpp

```diff
@@ -1,104 +1,120 @@
+#include <algorithm>
 #include <fstream>
 #include <iostream>
-#include <algorithm>
-#include <thread>
 #include <numeric>
 #include <random>
+#include <thread>
+#include <optional>
 
-#include "spdlog/spdlog.h"
 #include "mappable/mapper.hpp"
+#include "spdlog/spdlog.h"
+#include "payload_vector.hpp"
 
 #include "binary_freq_collection.hpp"
 #include "util/index_build_utils.hpp"
 #include "util/inverted_index_utils.hpp"
-#include "util/util.hpp"
 #include "util/progress.hpp"
-
-int main(int argc, const char** argv)
-{
-
-    using namespace pisa;
-
-    if (argc != 3 && argc != 4) {
-        std::cerr << "Usage: " << argv[0]
-                  << " <collection basename> <output basename> [ordering file]"
-                  << std::endl
-                  << "Ordering file is of the form <current ID> <new ID>"
-                  << std::endl;
-        return 1;
+#include "util/util.hpp"
+#include <CLI/CLI.hpp>
+
+int main(int argc, const char **argv) {
+
+  using namespace pisa;
+  std::string input_basename;
+  std::string output_basename;
+  std::optional<std::string> mapping_filename;
+  std::optional<std::string> documents_filename;
+  std::optional<std::string> reordered_documents_filename;
+
+  CLI::App app{"Shuffle document IDs."};
+  app.add_option("-c,--collection", input_basename, "Collection basename")
+      ->required();
+  app.add_option("-o,--output", output_basename, "Output basename");
+  app.add_option("--mapping-filename", mapping_filename, "Ordering file is of the form <current ID> <new ID>");
+  auto docs_opt = app.add_option("--documents", documents_filename, "Documents lexicon");
+  app.add_option("--reordered-documents", reordered_documents_filename, "Reordered documents lexicon")->needs(docs_opt);
+  CLI11_PARSE(app, argc, argv);
+
+  constexpr uint32_t seed = 1729;
```

Comment:
Should we add an option to override it from command line?

Reply:
Can you check now?